srtcore/buffer.cpp

```diff
@@ -224,6 +231,17 @@ void CSndBuffer::addBuffer(const char* data, int len, SRT_MSGCTRL& w_mctrl)
         m_iNextMsgNo = w_msgno;
     }
 
+    if (!w_srctime)
+    {
+        // Rewrite the actual sending time back into w_srctime
+        // so that the calling facilities can reuse it
+        w_srctime = count_microseconds(currtime.time_since_epoch());
+    }
+
+    // Intentionally dropping a possible sub-microsecond precision
+    // in case this timestamp is to be populated on other packets via msgctrl.src_time
+    const steady_clock::time_point timestamp = steady_clock::time_point() + microseconds_from(w_srctime);
```

Comment:
Isn't that simply enough if you make sure that the timestamp is forcefully set by the group sender function and leave this code as is? If a group sender sets the timestamp, it **will** be the same in every packet, no matter if you align it to microseconds or not (of course, if you make sure the time is aligned to microseconds in the group sender function).

Reply:
Yes, that was another option, and it would make the code of the group sender more clean and straightforward.
However, the concern is that we still provide a use case (not related to groups, just in general) where `src_time` is passed to the app via MSG_CTRL, and it can use it for some operations, including timestamping further packets. Therefore it might make sense to also use a microsecond accuracy internally.